test/jobs/queue_for_upload_job_test.rb

```diff
@@ -0,0 +1,134 @@
+    # Mock plan_upload_report to verify it's NOT called
+    ForemanInventoryUpload::Async::QueueForUploadJob.any_instance.expects(:plan_upload_report).never
+
+    assert_equal 'success', subject.result
+  end
+
+  test 'run method skips processing when content is disconnected via parameter' do
+    FileUtils.touch(report_path)
+    ForemanInventoryUpload::Async::QueueForUploadJob.any_instance.expects(:plan_upload_report).never
+
+    disconnected_subject = ForemanTasks.sync_task(ForemanInventoryUpload::Async::QueueForUploadJob, base_folder, report_file, organization.id, true)
+    assert_equal 'success', disconnected_subject.result
+
+    # File should still exist because plan returns early and run never executes
+    assert File.exist?(report_path), "Original file should still exist when disconnected via parameter"
+  end
+
+  test 'run method skips processing when subscription connection is disabled' do
+    Setting.stubs(:[]).with(:subscription_connection_enabled).returns(false)
+    ForemanInventoryUpload::Async::QueueForUploadJob.any_instance.expects(:plan_upload_report).never
+
+    assert_equal 'success', subject.result
+
+    # File should still exist because plan returns early and run never executes
+    assert File.exist?(report_path), "Original file should still exist when subscription connection disabled"
+  end
+
+  test 'run method processes file when connected' do
+    ForemanInventoryUpload::Async::QueueForUploadJob.any_instance.stubs(:plan_upload_report)
+
+    assert_equal 'success', subject.result
+
+    # Verify the file was moved
+    refute File.exist?(report_path), "Original file should be moved"
+    assert File.exist?(File.join(uploads_folder, report_file)), "File should exist in uploads folder"
+  end
+
+  test 'handles disconnection properly and does not move files' do
+    Setting.stubs(:[]).with(:subscription_connection_enabled).returns(false)
+    ForemanInventoryUpload::Async::QueueForUploadJob.any_instance.expects(:plan_upload_report).never
+
+    assert_equal 'success', subject.result, "Task should succeed when disconnected via subscription setting"
+
+    # Verify the file was NOT moved when disconnected (because plan returns early)
+    assert File.exist?(report_path), "Original file should still exist when disconnected via subscription setting"
+  end
+
+  test 'creates necessary folders and scripts when connected' do
+    ForemanInventoryUpload::Async::QueueForUploadJob.any_instance.stubs(:plan_upload_report)
+
+    assert_equal 'success', subject.result
+
+    # Verify the uploads folder was created
+    assert Dir.exist?(uploads_folder), "Uploads folder should be created"
+
+    # Verify the script file was created
+    script_path = File.join(uploads_folder, ForemanInventoryUpload.upload_script_file)
+    assert File.exist?(script_path), "Upload script should be created"
+  end
+
+  test 'logs contextual disconnected message for different disconnection scenarios' do
```

Comment:
We can split this into two cases. Then we can still use the setup in the beginning of the file, I think.